packages/text-input/src/TextInput.tsx

```diff
@@ -229,6 +241,46 @@ const interactionRingColor: Record<Mode, Record<'valid' | 'error', string>> = {
   },
 };
 
+interface SizeSet {
+  inputHeight: number;
+  inputText: number;
+  text: number;
+  lineHeight: number;
+}
+
+const sizeSets: Record<SizeVariantType, SizeSet> = {
+  [SizeVariantType.XSmall]: {
+    inputHeight: 22,
+    inputText: 12,
+    text: 14,
+    lineHeight: 20,
+  },
+  [SizeVariantType.Small]: {
+    inputHeight: 28,
+    inputText: 14,
+    text: 14,
+    lineHeight: 20,
+  },
```

Comment:
Looks like XS and S sizes have a bit too much left/right padding compared to Figma